### PR TITLE
[PFSP1-0003] Register Screen - UI

### PR DESCRIPTION
**Pull request by robbyaww**
***
## Description
Add UI register screen

## Related Links
List related PRs against other branches:

List | link
------ | ------
User Story | [link]()
Figma | [link]()
Issue | [link]()

## Todos
- [x] UI
- [ ] Coordinator
- [ ] Network
- [ ] Tests
- [ ] Documentation

## Screenshot / Video
Screenshot consisting of changes made on this branch.

### UI Changes
UI related changes will be shown here (can be either screenshot or video format)

### Network Status
Network related changes will be shown here (screenshot of request status)

### Test Coverage
Test coverage result for designated changes will be shown here (screenshot of coverage)

## Notes
Additional notes that you have either technical debt, blocker, etc.

### CODE REVIEW

File: Podfile.lock [hunk 1]
```diff
@@ -161,4 +161,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 7bcfd560fb0ce54bd4f2cf28164bbf33ea483123
 
-COCOAPODS: 1.9.1
```
Comment (inxudianz) [inline Podfile.lock:164]: Podfile must not be commited

File: MainApp/MainApp/Resources/Main.xcassets/Register/ic_google.imageset/Contents.json [hunk 2]
```diff
@@ -0,0 +1,23 @@
+{
+  "images" : [
```
Comment (inxudianz) [inline MainApp/MainApp/Resources/Main.xcassets/Register/ic_google.imageset/Contents.json]: images specific for a certain feature must be put inside its own xcassets inside Feature/<name>/Resources/<name>.xcasset

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+    
+    private func isValidEmail(_ email: String) -> Bool {
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+        
+        let emailPred = NSPredicate(format:"SELF MATCHES %@", emailRegEx)
+        return emailPred.evaluate(with: email)
+    }
+    
+    private func isValidPassword(_ password: String) -> Bool {
+        let PasswordRegEx = "^(?=.*[A-Z].*[A-Z])(?=.*[!@#$&*])(?=.*[0-9].*[0-9])(?=.*[a-z].*[a-z].*[a-z]).{8}$"
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Register/RegisterViewController.swift]: Business logic should be in viewModel

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+    
+    private func isValidPhone(_ phone: String) -> Bool {
+        let phoneRegex = "^[0-9+]{0,1}+[0-9]{5,16}$"
+        let phoneTest = NSPredicate(format: "SELF MATCHES %@", phoneRegex)
+        return phoneTest.evaluate(with: phone)
+    }
+    
+    private func isValidEmail(_ email: String) -> Bool {
+        let emailRegEx = "[A-Z0-9a-z._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,64}"
+
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Register/RegisterViewController.swift]: Business logic should be in viewModel

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+        } else {
+            viewLabelError.isHidden = false
+            labelError.text = "Password does't match"
+            tfConfirmPassword.layer.borderColor = #colorLiteral(red: 1, green: 0.05882352941, blue: 0, alpha: 1)
+        }
+    }
+    
+    private func isValidPhone(_ phone: String) -> Bool {
+        let phoneRegex = "^[0-9+]{0,1}+[0-9]{5,16}$"
+        let phoneTest = NSPredicate(format: "SELF MATCHES %@", phoneRegex)
```
Comment (inxudianz) [inline MainApp/MainApp/Feature/Register/RegisterViewController.swift]: Business logic should be in viewModel

**Review by inxudianz — CHANGES_REQUESTED** [review]

File: Component/Component/Button/FamiliButton/FamiliButton.swift [hunk 6]
```diff
@@ -96,35 +96,35 @@ How to use:
...
-    private func setViewPrimary() {
+    public func setViewPrimary() {
         self.backgroundColor = #colorLiteral(red: 0, green: 0.7647058824, blue: 1, alpha: 1)
         self.tintColor = #colorLiteral(red: 1.0, green: 1.0, blue: 1.0, alpha: 1.0)
         setComponentButton()
     }
     
     /// Set secondary view
-    private func setViewSecondary() {
+    public func setViewSecondary() {
```
Comment (inxudianz) [inline Component/Component/Button/FamiliButton/FamiliButton.swift]: undocumented changes in unrelated PR

**Review by inxudianz — CHANGES_REQUESTED** [review]

File: MainApp/MainApp/Coordinator/Feature/Register/RegisterCoordinator.swift [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+    required init(navigationController: UINavigationController) {
+        self.navigationController = navigationController
+    }
+    
+    func start() {
+        let vc = RegisterViewController()
+        let vm = RegisterViewModel()
+        vm.coordinator = self
+        vc.viewModel = vm
+        //vm.view = vc
```
Comment (inxudianz) [inline MainApp/MainApp/Coordinator/Feature/Register/RegisterCoordinator.swift]: why commented

**Review by inxudianz — CHANGES_REQUESTED** [review]

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift (hunk 5 above)
Comment (robbyaww) [reply MainApp/MainApp/Feature/Register/RegisterViewController.swift]: already moved to the view model

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift (hunk 4 above)
Comment (robbyaww) [reply MainApp/MainApp/Feature/Register/RegisterViewController.swift]: already moved to the viewModel

File: MainApp/MainApp/Feature/Register/RegisterViewController.swift (hunk 3 above)
Comment (robbyaww) [reply MainApp/MainApp/Feature/Register/RegisterViewController.swift]: already moved to the viewModel

File: MainApp/MainApp/Resources/Main.xcassets/Register/ic_google.imageset/Contents.json (hunk 2 above)
Comment (robbyaww) [reply MainApp/MainApp/Resources/Main.xcassets/Register/ic_google.imageset/Contents.json]: has created new file register.xcasset

File: Component/Component/Button/FamiliButton/FamiliButton.swift (hunk 6 above)
Comment (robbyaww) [reply Component/Component/Button/FamiliButton/FamiliButton.swift]: has changed to previous commit

**Pull request closed without merging**